tests/rptest/tests/pandaproxy_test.py

```diff
@@ -1238,3 +1240,90 @@ def test_password_change(self):
         # teardown process in RedpandaService relies on the superuser
         admin.update_user(super_username, super_password, super_algorithm)
+
+
+class PandaProxyClientStopTest(PandaProxyEndpoints):
+    username = 'red'
+    password = 'panda'
+    algorithm = 'SCRAM-SHA-256'
+
+    topics = [TopicSpec()]
+
+    def __init__(self, context):
+
+        security = SecurityConfig()
+        security.enable_sasl = True
+        security.endpoint_authn_method = 'sasl'
+        security.pp_authn_method = 'http_basic'
+
+        super(PandaProxyClientStopTest, self).__init__(
+            context,
+            security=security,
+            pp_keep_alive=60000 * 5,  # Time in ms
+            pp_cache_max_size=1)
+
+    @cluster(num_nodes=3)
+    def test_client_stop(self):
+        super_username, super_password, super_algorithm = self.redpanda.SUPERUSER_CREDENTIALS
+        rpk = RpkTool(self.redpanda)
+
+        o = rpk.sasl_create_user(self.username, self.password, self.algorithm)
+        self.logger.debug(f'Sasl create user {o}')
+
+        # Only the super user can add ACLs
+        o = rpk.sasl_allow_principal(f'User:{self.username}', ['all'], 'topic',
+                                     self.topic, super_username,
+                                     super_password, super_algorithm)
+        self.logger.debug(f'Allow all topic perms {o}')
+
+        # Issue some request so that the client cache holds a single
+        # client for the super user
+        result_raw = self._get_topics(auth=(super_username, super_password))
+        assert result_raw.status_code == requests.codes.ok
+        assert result_raw.json()[0] == self.topic
+
+        data = '''
+        {
+            "records": [
+                {"value": "dmVjdG9yaXplZA==", "partition": 0},
+                {"value": "cGFuZGFwcm94eQ==", "partition": 1},
+                {"value": "bXVsdGlicm9rZXI=", "partition": 2}
+            ]
+        }'''
+
+        import time
+
+        def _produce_req(username, userpass, timeout_sec=30):
+            start = time.time()
+            stop = start + timeout_sec
+            while time.time() < stop:
+                try:
+                    self.logger.info(
+                        f"Producing to topic: {self.topic}, User: {username}")
+                    produce_result_raw = self._produce_topic(self.topic,
+                                                             data,
+                                                             auth=(username,
+                                                                   userpass))
+                    self.logger.debug(
+                        f"Producing to topic: {self.topic}, User: {username}, Result: {produce_result_raw.status_code}"
+                    )
+                    assert produce_result_raw.status_code == requests.codes.ok
+                except:
+                    return requests.codes.internal_server_error
```

Comment:
Is it only the assert that gets us to the except?  Why not something like:
```suggestion
                    if produce_result_raw.status_code != requests.codes.ok
                        return produce_result_raw.status_code
```

Reply:
I also wanted to catch any errors from the call to `_produce_topic` but now that I'm taking another look I realize that method doesn't throw. I'll make the change.

Reply:
Done.